in-development/mission_3774_sap_task_center/main.tf

```diff
@@ -0,0 +1,134 @@
+# Setup of names in accordance to naming convention
+# ------------------------------------------------------------------------------------------------------
+resource "random_uuid" "uuid" {}
+
+locals {
+  random_uuid               = random_uuid.uuid.result
+  project_subaccount_domain = "discoverycenter-tf-sap-ms-${local.random_uuid}"
+  project_subaccount_cf_org = substr(replace("${local.project_subaccount_domain}", "-", ""), 0, 32)
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Creation of subaccount
+# ------------------------------------------------------------------------------------------------------
+resource "btp_subaccount" "project" {
+  name      = var.subaccount_name
+  subdomain = local.project_subaccount_domain
+  region    = lower(var.region)
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Assignment of users as sub account administrators
+# ------------------------------------------------------------------------------------------------------
+resource "btp_subaccount_role_collection_assignment" "subaccount-admins" {
+  for_each             = toset("${var.subaccount_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Administrator"
+  user_name            = each.value
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Assignment of users as sub account service administrators
+# ------------------------------------------------------------------------------------------------------
+resource "btp_subaccount_role_collection_assignment" "subaccount-service-admins" {
+  for_each             = toset("${var.subaccount_service_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Service Administrator"
+  user_name            = each.value
+}
+
+###############################################################################################
+# Creates a cloud foundry environment in a given account
+###############################################################################################
+// and the dedicted target landscape cf-us10
+resource "btp_subaccount_environment_instance" "cloudfoundry" {
+  subaccount_id    = btp_subaccount.project.id
+  name             = var.cf_org_name
+  environment_type = "cloudfoundry"
+  service_name     = "cloudfoundry"
+  landscape_label  = null_resource.cache_target_environment.triggers.label
+  plan_name        = "standard"
+  parameters = jsonencode({
+    instance_name = var.cf_org_name
+  })
+}
+
+###############################################################################################
+# Create the Cloud Foundry space
+###############################################################################################
+resource "cloudfoundry_space" "space" {
```

Comment:
Setup of CF in setup won't work due to extension landscapes (unknown landscape label). This must be split into two steps for QAS setup.